provider/lxd/environ_instance.go

```diff
@@ -135,6 +135,11 @@ func (env *environ) AdoptResources(ctx context.ProviderCallContext, controllerUU
 	qualifiedKey := lxd.UserNamespacePrefix + tags.JujuController
 	for _, instance := range instances {
 		id := instance.Id()
+		// TODO (manadart 2018-06-27) This is a smell.
+		// Everywhere else, we update the container config on a container and then call WriteContainer.
+		// If we added a method directly to environInstance to do this, we wouldn't need this
+		// implementation of UpdateContainerConfig at all, and the container representation we are
+		// holding would be consistent with that on the server.
```

Comment:
So this is one of the cases where we are explicitly mutating in place because we are moving from one controller to another controller (this is called during Migration, which is a fairly different-from-normal operation).

Are the other ones cases where we are doing live migration vs just setting the values before starting?

Reply:
I will check this, but I don't think it matters *that* much. I am pretty sure all the usages are on very short-lived objects - it just seemed more pure not to have the representations diverge.